content/articles/2018-11-tibble-2.0.0-pre-announce.Rmd

```diff
@@ -0,0 +1,222 @@
+
+```{r}
+# Alternative: use new `.name_repair` argument to permit dysfunctional names
+m <- cov(unname(iris[-5]))
+as_tibble(m, .name_repair = "minimal")
+
+# Alternative: use new `.name_repair` argument to fix dysfunctional names
+m <- cov(unname(iris[-5]))
+as_tibble(m, .name_repair = "unique")
+```
+
+What is the motivation for this increased attention to column names? The tibble package is offering stronger encouragement for names where each column can be identified by name and, preferably, without having to resort to backticks. Column names that don't meet these requirements are still allowed, but the user needs to permit them explicitly.
+
+After all, there are scenarios where problematic names should be tolerated. For example, after importing data, the user might need to inspect the data in order to determine which columns to keep. Or perhaps the column names contain data that is about to be converted to a proper variable with `gather()`.
+
+The `tibble()` constructor and the `as_tibble()` generic now support a new `.name_repair` argument that covers most use cases:
+
+- `"minimal"`: No name repair or checks, beyond basic existence.
+- `"unique"`: Make sure names are unique and not empty.
+- `"check_unique"`: (default value), no name repair, but check they are `unique`.
+- `"universal"`: Make the names `unique` and syntactic.
+- a function: apply custom name repair (e.g., `.name_repair = make.names` or `.name_repair = ~make.names(., unique = TRUE)` for names in the style of base R).
+
+See `` ?`name-repair` `` for more details.
+
+Packages that are in the business of making tibbles may even want to expose the `.name_repair` argument and pass it through to tibble. For example, this is the approach planned for [readxl](https://readxl.tidyverse.org), which reads rectangular data out of Excel workbooks.
+
+## Deprecation of `validate` in `as_tibble()`
+
+In tibble v1.4.2, `as_tibble()` has a `validate` argument, but its default behaviour value was inconsistent across different methods and there was no equivalent argument for `tibble()`. The `validate` argument is now soft-deprecated and its use will trigger a message, once per session. The `validate` argument will eventually be removed, but for now it can be used jointly with the new `.name_repair` argument (without even triggering a message). This is possible, because fortunately the `.name_repair` argument to `as_tibble()` is ignored in tibble v1.4.2.
+
+Here's what `validate` does in the development version of tibble for tibbles, data frames, and matrices, along with suggested alternatives.
+
+### Tibbles and data frames
+
+The default was `validate = FALSE` for tibbles and `validate = TRUE` for data frames.
+Code that worked before for tibbles can now throw unexpected errors if the resulting tibble has problematic names.
+Use the default instead of `validate = TRUE`, and `.name_repair = "minimal"` in addition to `validate = FALSE`.
```

Comment:
I'm a little confused by how this line connects to the ones above it. Is this the recommendation for what one should do now?